Fix to allow repo replication when using non-standard port for core

nickhyoti opened the pull request:
Use CORE_LOCAL_URL instead of hardcoded setting

Thank you for contributing to Harbor!

# Comprehensive Summary of your change
In our environment we are using a non-standard port setting for Core.  
Instead of 8080, we are using 8000.
When we try and setup repository replication we get a message 

Failed to create adapter for destination registry http://core:
Get "http://127.0.0.1:8080/api/version": dial tcp http://127.0.0.1:8080:connect: connection refused

This is because there is a hardcoded port in the function GetURL.
There is already an environment variable defined which has the correct URL for core, so use that instead

# Issue being fixed
Fixes unable to setup repo replication when using non-standard port on core

Please indicate you've done the following:
- [X] Well Written Title and Summary of the PR
- [X] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [X] Accepted the DCO. Commits without the DCO will delay acceptance.
- [X] Made sure tests are passing and test coverage is added if needed.
- [X] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

nickhyoti committed:
Fix to allow repo replication when using non-standard port for core

Use CORE_LOCAL_URL instead of hardcoded setting

Signed-off-by: Nick Hindley <nicholas.hindley@yoti.com>

stonezdj committed:
Refine the sql to query non empty repositories (#22269)

use exists instead of in condition

Signed-off-by: stonezdj <stone.zhang@broadcom.com>

bupd reviewed (changes requested):
@nickhyoti Thanks for your contributions 

I have added my suggestion for improvements.

nickhyoti committed:
Fixed compilation error
Added check for environment variable not set

Signed-off-by: Nick Hindley <nicholas.hindley@yoti.com>

stonezdj commented:
How do you specify another port other than 8080? How do you deploy the Harbor? docker-compose or kubernetes?

dependabot[bot] committed:
chore(deps): bump github.com/beego/beego/v2 from 2.3.6 to 2.3.8 in /src (#22282)

Bumps [github.com/beego/beego/v2](https://github.com/beego/beego) from 2.3.6 to 2.3.8.
- [Release notes](https://github.com/beego/beego/releases)
- [Changelog](https://github.com/beego/beego/blob/master/CHANGELOG.md)
- [Commits](https://github.com/beego/beego/compare/v2.3.6...v2.3.8)

---
updated-dependencies:
- dependency-name: github.com/beego/beego/v2
  dependency-version: 2.3.8
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nickhyoti committed:
Formatting fixed
Additional check for environment variable not defined

Signed-off-by: Nick Hindley <nicholas.hindley@yoti.com>

nickhyoti commented on src/pkg/reg/adapter/harbor/base/client.go:
```diff
@@ -147,7 +147,13 @@ func (c *Client) GetURL() string {
+
+	// get the URL from the environment so can use non-standard port
+        url := os.Getenv("CORE_LOCAL_URL")
+	if (url == "") {
```
Thanks - added condition

nickhyoti committed:
Merge branch 'main' into Non_standard_port_fix

Signed-off-by: Nick Hindley <62100881+nickhyoti@users.noreply.github.com>

nickhyoti committed:
Merge branch 'main' into Non_standard_port_fix
